homeassistant/components/airvisual/__init__.py

```diff
@@ -108,6 +108,7 @@ def async_get_cloud_coordinators_by_api_key(
         for entry_id, attrs in hass.data[DOMAIN].items()
         if (entry := hass.config_entries.async_get_entry(entry_id))
         and entry.data.get(CONF_API_KEY) == api_key
+        and attrs.get(DATA_COORDINATOR) is not None
```

Comment:
You can avoid a double dict access with a walrus:

```python
return [
    coordinator
    for entry_id, attrs in hass.data[DOMAIN].items()
    if (entry := hass.config_entries.async_get_entry(entry_id))
    and (coordinator := attrs.get(DATA_COORDINATOR))
    and entry.data.get(CONF_API_KEY) == api_key
]
```

The order also adds a short-circuit that doesn't bother checking the API key if the coordinator doesn't exist.

Reply:
clever !

Reply:
I put the suggestion in pull request

Reply:
Please can you check (I've tested it's ok)